update migration requirement for in-tree vSphere volumes (#35551)
kubernetes/website: content/en/docs/concepts/storage/volumes.md

{{< feature-state for_k8s_version="v1.19" state="beta" >}}

> The `CSIMigration` feature for `vsphereVolume`,  is enabled by default as of Kubernetes v1.25.

Reviewer: ```suggestion
The `CSIMigration` feature for `vsphereVolume` is enabled by default as of Kubernetes v1.25.
```

Reviewer 2: This doesn't sound quite right: the text is mixing up `CSIMigration` (a feature gate which is enabled by default) on the one hand, and _CSI migration for `vsphereVolume` storage_ (a specific case of that feature) on the other. Readers looking at this might thing that Kubernetes v1.24 didn't have `CSIMigration` enabled.

In this paragraph the only mention of `CSIMigration` should be to point out that you cannot benefit from the automatic migration of vSphere volumes if you have **disabled** either `CSIMigration` or `CSIMigrationvSphere`.

Author: @sftim @divya-mohan0209 

I am changing this line to following

```
The `CSIMigrationvSphere` feature for `vsphereVolume` is enabled by default as of Kubernetes v1.25.
All plugin operations from the in-tree `vspherevolume` will be redirected to the `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver. 
```

> All plugin operations from the in-tree `vspherevolume` will be redirected to the `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver. 

Reviewer: ```suggestion
All plugin operations from the in-tree `vsphereVolume` will be redirected to the `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver. 
```

Reviewer 2: This is only true provided that you don't force `CSIMigration` to off, and you also don't force `CSIMigrationvSphere` to off   (or maybe your cloud provided made that choice for you) - and we should make that clear.

Author: both `CSIMigration` and `CSIMigrationvSphere` are enabled in K8s 1.25, so do we need to mention that we can disable `CSIMigration` and `CSIMigrationvSphere` features?

Author: @sftim how about following message

> All plugin operations from the in-tree `vspherevolume` will be redirected to the `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver unless feature gates `CSIMigration` or `CSIMigrationvSphere` are disabled.

[vSphere CSI driver](https://github.com/kubernetes-sigs/vsphere-csi-driver)

> must be installed on the cluster. You can find additional advice on how to migrate in VMware's

Reviewer: The sentence on migration doesn't sound right because we're not specifying what we're migrating to or what we're migrating from. 

```suggestion
must be installed on the cluster. You can find additional advice on how to migrate in VMware's
```

Author: changed it to 

> You can find additional advice on how to migrate in-tree `vsphereVolume` in VMware's
documentation page`

documentation page [Migrating In-Tree vSphere Volumes to vSphere Container Storage Plug-in](https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/2.0/vmware-vsphere-csp-getting-started/GUID-968D421F-D464-4E22-8127-6CB9FF54423F.html).

Kubernetes v{{< skew currentVersion >}} requires that you are using vSphere 7.0u2 or later
in order to migrate to the out-of-tree CSI driver.

vSphere releases less than 7.0u2 are not supported for in-tree vSphere volume as of Kubernetes v1.25. 
Please consider upgrading vSphere (both ESXi and vCenter)  to 7.0u2 or above.

> 

Reviewer 2: These two paragraphs seem to say the same thing?

Author: removed one of them.

{{< feature-state for_k8s_version="v1.19" state="beta" >}}

> The `CSIMigrationvSphere` feature for `vsphereVolume` is enabled by default as of Kubernetes v1.25.

Reviewer 3: Does this mean that the feature gate doc should be updated to reflect this change?

Reviewer 4: You are right. It's covered by this other PR: https://github.com/kubernetes/website/pull/35685

{{< feature-state for_k8s_version="v1.19" state="beta" >}}

The `CSIMigrationvSphere` feature for `vsphereVolume` is enabled by default as of Kubernetes v1.25.

> All plugin operations from the in-tree `vspherevolume` will be redirected to the `csi.vsphere.vmware.com` {{< glossary_tooltip text="CSI" term_id="csi" >}} driver unless feature gates `CSIMigration` or `CSIMigrationvSphere` are disabled.

Reviewer 2: I might be wrong, but I think that actually the `CSIMigration` feature gate will be graduated (locked to `true`) for v1.25. Check with SIG Storage folks to find out the actual situation with the related CSI driver migration feature graduations.

Author: checked https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.25.md#feature-2
`CSIMigration` is locked in 1.25.

[vSphere CSI driver](https://github.com/kubernetes-sigs/vsphere-csi-driver)
must be installed on the cluster. You can find additional advice on how to migrate in-tree `vsphereVolume` in VMware's
documentation page [Migrating In-Tree vSphere Volumes to vSphere Container Storage Plug-in](https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/2.0/vmware-vsphere-csp-getting-started/GUID-968D421F-D464-4E22-8127-6CB9FF54423F.html).

> Kubernetes v{{< skew currentVersion >}} requires that you are using vSphere 7.0u2 or later
> in order to migrate to the out-of-tree CSI driver.
> 
> vSphere releases less than 7.0u2 are not supported for in-tree vSphere volume as of Kubernetes v1.25. 
> Please consider upgrading vSphere (both ESXi and vCenter)  to 7.0u2 or above.

Reviewer 2: ```suggestion
As of Kubernetes v1.25, vSphere releases less than 7.0u2 are not supported for the
(deprecated) in-tree vSphere storage driver. You must run vSphere 7.0u2 or later
in order to either continue using the deprecated driver, or to migrate to
the replacement CSI driver.
```

This clarifies that the in-tree driver is deprecated, and ties in to the following paragraph about reading the right docs for the version of Kubernetes you're actually running.
